endpoint/endpoint.go

```diff
@@ -34,17 +42,36 @@ type Endpoint struct {
 	Target string
 	// RecordType type of record, e.g. CNAME, A, TXT etc
 	RecordType string
+	// TTL for the record
+	RecordTTL TTL
 	// Labels stores labels defined for the Endpoint
 	Labels map[string]string
 }
 
 // NewEndpoint initialization method to be used to create an endpoint
 func NewEndpoint(dnsName, target, recordType string) *Endpoint {
+	return NewEndpointWithTTLValue(dnsName, target, recordType, nil)
```

Comment:
ok I double checked. TTL=0 is not defined in the standard, so we can use it to mean "not configured and thus use default". :+1: